File: day 02/day2.cs
using System;
using System.IO;

namespace AdventProject
{
  class AdventOfCode
  {
    static void Main(string[] args)
    {
      int solutionPart1 = 0, solutionPart2 = 0;
      int lowest, highest;
      char passwordPolicy;
      String password, numbers;
      string inputFile = Directory.GetParent(Directory.GetCurrentDirectory()).Parent.Parent.FullName + @"\in.txt";
      StreamReader cin = new StreamReader(inputFile);

      string inputLine;
      while (!cin.EndOfStream)
      {
        inputLine = cin.ReadLine();

        int index = inputLine.IndexOf('-');
        int index2 = inputLine.IndexOf(' ');

        lowest = int.Parse(inputLine.Substring(0, index));
        highest = int.Parse(inputLine.Substring(index + 1, index2 - index - 1));

        passwordPolicy = inputLine[index2 + 1];
        password = inputLine.Substring(index2 + 4, inputLine.Length - index2 - 4);

        int count = 0;
        foreach (char p in password)
        {
          if (p == passwordPolicy)
          {
            count++;
          }
        }

        if (count >= lowest && count <= highest)
        {
          solutionPart1++;
        }

        if (password[lowest - 1] == passwordPolicy && password[highest - 1] != passwordPolicy
            || password[lowest - 1] != passwordPolicy
                && password[highest - 1] == passwordPolicy)
        {
          solutionPart2++;
        }
      }

      Console.WriteLine("Part 1: " + solutionPart1);
      Console.WriteLine("Part 2: " + solutionPart2);

    }
  }
}

File: day 02/day2.cpp
#define _CRT_SECURE_NO_WARNINGS
#include <iostream>
#include <string>

using namespace std;

int main()
{
  freopen("in.txt", "r", stdin);
  freopen("out.txt", "w", stdout);

  int solutionPart1 = 0, solutionPart2 = 0;

  int lowest, highest;
  char passwordPolicy;
  string password;

  char delimiter;

  while (cin >> lowest >> delimiter >> highest >> passwordPolicy >> delimiter >> password)
  {
    int count = std::count(password.begin(), password.end(), passwordPolicy);

    if (count >= lowest && count <= highest)
    {
      solutionPart1++;
    }

    if (password[lowest - 1] == passwordPolicy && password[highest - 1] != passwordPolicy || password[lowest - 1] != passwordPolicy && password[highest - 1] == passwordPolicy)
    {
      solutionPart2++;
    }
  }

  cout << "Part 1: " << solutionPart1 << endl;
  cout << "Part 2: " << solutionPart2 << endl;

  return 0;
}
